### PR TITLE
ath79: move ubnt-xm 64M RAM boards back to generic

### PR DESCRIPTION
**Pull request by Djfe**
return ubnt_rocket-m and ubnt_powerbridge-m back to ath79-generic
They have enough RAM-ressources to not be considered as tiny.

This reverts the commit f4415f7635164ec07ddc22f56df93555804b5767 partially

_________________________________

The gluon devs believe the Ubiquiti Rocket M2/M5 and PowerBridge M2/M5 devices where moved to tiny by accident.
They offer enough RAM ressources to still run kernels 5.10 and upwards without any issues.

@PolynomialDivision Please take a look at my PR, if you can spot any issues.
This was mainly a revert and shouldn't introduce any changes whatsoever (compared to OpenWrt 22.03.01
I would appreciate your feedback very much :)

I'd also like to know whether the calibration-data for these two boards appears incorrect.
Now would be a great time to fix those as well. You did so a couple month back for some boards that shared size 0x1000 calibration data before your change.
https://github.com/openwrt/openwrt/commit/a14170b6e9c8bab493c5e0e7689405a8a0675e8a

~~I still have to build-test this.~~ went successfully
I can provide a run-test for new calibration data on the rocket-m5 if required. (the other devices I don't have any access to)

### CODE REVIEW

**Comment by Djfe** [comment]
I just updated/improved the Wiki to reflect the ath79-tiny target and other stuff
https://openwrt.org/toh/views/toh_extended_all?dataflt%5BVersions*%7E%5D=XM
And I already set the two devices from this PR to generic in preparation of this commit.
The rocket got a new ToH entry (it never had one)

**Comment by Djfe** [comment]
Build test of both devices went successfully. Ready for review and then merge

**Comment by PolynomialDivision** [comment]
You are right that we can move them back to generic.

**Comment by Djfe** [comment]
CPU Info of a Rocket M2 (OpenWrt 2019.07, ar71xx):
`Atheros AR7241 rev 1` (probably the cpu class in Linux?)
https://paste.tecff.de/?033f9f0db1df2787#fCNkM/3Y/LQYmf0ZlQnLEhGNdZP3UAIcV8+qxbcFujg=

I discovered a better fcc id than `SWX-M2`

This one has internal images of **M2**: WiFi is clearly using a AR9342-DL3A
https://fccid.io/SWX-RM2W/Internal-Photos/Internal-Photos-4058636
This one has internal images of **M5**: WiFi appears to be using a AR9281-AL1E (AL1E lead me to believe it's ending in 1 or 5)
https://fccid.io/SWX-RM5/Internal-Photos/Internal-Photos-1611198

I created another ToH specifically for the Rocket M5.
Both appear to be using this switch: AR8032-B
If the cpu isn't correct anymore (AR9342 appears to be the soc, not AR7241), then pls let me know.

TL;DR I have one question for @PolynomialDivision
Is it possible that we need to adjust the caldata for them? (like you did here for other devices: https://github.com/openwrt/openwrt/commit/a14170b6e9c8bab493c5e0e7689405a8a0675e8a )
The rocket isn't in that commit because you moved it to tiny earlier.

**Comment by Djfe** [comment]
This is the art partition of a rocket m2 (called eeprom in openwrt 2019)
https://upthis.de/dl/u4ql6o/rocket_m2_eepromdd.bin

**Comment by rotanid** [comment]
@PolynomialDivision would be great if you could comment on the question by @Djfe regarding the caldata :)

File: target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom [hunk 1]
```diff
@@ -113,6 +113,10 @@ case "$FIRMWARE" in
 	tplink,tl-wr842n-v1)
 		caldata_extract "art" 0x1000 0x3e0
 		;;
+	ubnt,powerbridge-m|\
+	ubnt,rocket-m)
+		caldata_extract "art" 0x1000 0x1000
```
Comment (DragonBluep) [inline target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:118]: ART size 0x1000 is suspect. If you open art partition backup file with hex editor, you will see a clear boundary between art and redundant data. And it's a good chance to convert them to nvmem format if you can test it.

**Comment by Djfe** [comment]
I'll report back again. This change will affect a number of devices. though most are tiny and by now disabled.
I can ask a volunteer we found to test this on a rocket m5. (we have no volunteer for the power bridge-m)
I don't have access to this hardware myself

**Comment by PolynomialDivision** [comment]
@Djfe Okay, I will merge this PR now and please add another PR converting it when you tested everything. :)

**Comment by PolynomialDivision** [comment]
Merged in https://github.com/openwrt/openwrt/commit/9e86a96af5ecea6e73fa463d6cfa4408e2b49d35

**Comment by Djfe** [comment]
Thanks, much appreciated.
Here's a reupload of the art/eeprom partition
https://gigamove.rwth-aachen.de/de/download/7ed55b6cc34ae9e14d62c24cf2c7a879

It contains three mac addresses:
- two right at the beginning for ethernet (like defined in the dts)
- one at <0x120C 0x6> (part of the caldata)

Could you help me out with the size, pls?
It could be 0x35C, 0xA17, 0xEB8

Looking at art as a whole: Is the data at offset 0x9000 important?

**Comment by DragonBluep** [comment]
It's 0xeb8.
I find a pattern. For ath9k NIC:
||older than ar9287|ar9287|newer than ar9287|
|-|-|-|-|
|art size|0xeb8|0x3d8|0x440|

**Comment by Djfe** [comment]
great :) 
which ones are 0x3e0 then?
https://github.com/openwrt/openwrt/blob/cd82fc2235d4a11fc33e5b590d8f3bc6a7b13b78/target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom#L114C14-L114C14

or should that be corrected to 0x3d8?

**Comment by DragonBluep** [comment]
> or should that be corrected to 0x3d8?

Yes, refer to https://github.com/openwrt/openwrt/pull/11555#discussion_r1082739334.

**Comment by DragonBluep** [comment]
Update:
From ath9k source code https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/drivers/net/wireless/ath/ath9k/eeprom.c?h=next-20230719#n671
||older than ar9287|ar9285/ar9271|ar9287|newer than ar9287|
|-|-|-|-|-|
|art size|0xeb8|0x1f8?|0x3d8|0x440|

**Comment by PolynomialDivision** [comment]
@Djfe Do you care that this PR also land in 23.05?

**Comment by Djfe** [comment]
yes, and 22.03
We are going to be running on that release until Gluon has been tested enough on the new release :)

**Comment by rotanid** [comment]
backport to 23.05 was committed on 2023-08-11:
ac34f6469ecf07cdeb32f65788192262fd8e0100

someone planning to backport it to 22.03 ?
